lib/ansible/modules/cloud/amazon/aws_codedeploy_application.py

```diff
@@ -0,0 +1,433 @@
+  application_name: TestApplication
+  deployment_group: test_application_group
+  deployment_style:
+    deploymentType: "IN_PLACE"
+    deploymentOption: "WITHOUT_TRAFFIC_CONTROL"
+  service_role_arn: "arn:aws:iam::aws:policy/service-role/AWSCodeDeployRole"
+
+- aws_codedeploy_application:
+  application_name: TestApplication
+  deployment_group: test_application_group
+  deployment_style:
+    deploymentType: "IN_PLACE"
+    deploymentOption: "WITHOUT_TRAFFIC_CONTROL"
+  service_role_arn: "arn:aws:iam::aws:policy/service-role/AWSCodeDeployRole"
+  auto_scaling_groups: Test-Application-ASG
+  deployment_config: CodeDeployDefault.AllAtOnce
+
+- aws_codedeploy_application:
+  application_name: TestApplication
+  new_application_name: TestApplicationV1
+  deployment_group: test_application_group
+  new_deployment_group: test_application_v1
+  deployment_style:
+    deploymentType: "IN_PLACE"
+    deploymentOption: "WITH_TRAFFIC_CONTROL"
+  service_role_arn: "arn:aws:iam::aws:policy/service-role/AWSCodeDeployRole"
+  deployment_config: CodeDeployDefault.OneAtATime
+  load_balance_info:
+    elbInfoList:
+      - MyClassicLoadBalancer
+'''
+
+RETURN = '''
+application:
+  description: Information regarding the CodeDeploy application.
+  type: complex
+  returned: success
+  contains:
+    application_id:
+      description: The application ID.
+      type: string
+      sample: "81293494"
+    application_name:
+      description: The application name.
+      type: string
+      sample: "TestApplication"
+    create_time:
+      description: The time at which the application was created.
+      type: datetime
+      sample: 2017-05-30 010:00:00-05:00
+    linked_to_github:
+      description: True if the user has authenticated with GitHub for the specified application; otherwise, false.
+      type: boolean
+      sample: false
+'''
+
+from ansible.module_utils.basic import AnsibleModule
+from ansible.module_utils.ec2 import boto3_conn, get_aws_connection_info, ec2_argument_spec, camel_dict_to_snake_dict
+import traceback
+
+try:
+    import boto3
+    from botocore.exceptions import ClientError, NoCredentialsError
+    HAS_BOTO3 = True
+except ImportError:
+    HAS_BOTO3 = False
+
+
+def get_codedeploy_application(client, name, module):
+    """Get the details of the CodeDeploy application."""
+    try:
+        return client.get_application(applicationName=name)
+    except (ClientError, NoCredentialsError) as e:
+        if e.response['Error']['Code'] == 'ApplicationDoesNotExistException':
+            return None
+        else:
+            module.fail_json(msg=e.message, **camel_dict_to_snake_dict(e.response))
+    return None
+
+
+def get_codedeploy_application_deployment_group(client, name, dg, module):
+    """Get the details of the CodeDeploy deployment group."""
+    try:
+        return client.get_deployment_group(applicationName=name, deploymentGroupName=dg)
+    except (ClientError, NoCredentialsError) as e:
```

Comment:
It's worth catching BotoCoreError rather than NoCredentialsError (as that will catch ParameterValidation exceptions too).